src/TwigComponent/tests/Unit/ComponentAttributesTest.php

```diff
@@ -87,4 +87,22 @@ public function testCanAddStimulusController(): void
             'data-foo-name-value' => 'ryan',
         ], $attributes->all());
     }
+
+    public function testBooleanBehaviour(): void
+    {
+        $attributes = new ComponentAttributes(['disabled' => true]);
+
+        $this->assertSame(['disabled' => true], $attributes->all());
+        $this->assertSame(' disabled', (string) $attributes);
+
+        $attributes = new ComponentAttributes(['disabled' => false]);
+
+        $this->assertSame(['disabled' => false], $attributes->all());
+        $this->assertSame('', (string) $attributes);
+
+        $attributes = new ComponentAttributes(['disabled' => null]);
+
+        $this->assertSame(['disabled' => null], $attributes->all());
+        $this->assertSame(' disabled', (string) $attributes);
```

Comment:
I agree that this is wrong (it should have the same behavior as `false` - but it should be allowed). The other 2 cases are correct.

Any thoughts on gently (from a BC pov) adjusting null's behavior so it's the same as false?

Reply:
How do you suggest we do this?

Reply:
Let's "punt" on this. For now, let's trigger a deprecation saying that `null` isn't allowed.

Then in the next major, `null` could throw an exception. Or perhaps later we add a way to "opt into" allowing null with the new behavior.